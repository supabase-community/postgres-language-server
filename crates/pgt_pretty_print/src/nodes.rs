use crate::{
    TokenKind,
    emitter::{EventEmitter, ToTokens},
};

impl ToTokens for pgt_query::Node {
    fn to_tokens(&self, e: &mut EventEmitter) {
        if let Some(node) = &self.node {
            node.to_tokens(e);
        }
    }
}

impl ToTokens for pgt_query::protobuf::node::Node {
    fn to_tokens(&self, e: &mut EventEmitter) {
        match self {
            pgt_query::protobuf::node::Node::SelectStmt(stmt) => stmt.as_ref().to_tokens(e),
            _ => {
                unimplemented!("Node type {:?} not implemented for to_tokens", self);
            }
        }
    }
}

impl ToTokens for pgt_query::protobuf::SelectStmt {
    fn to_tokens(&self, e: &mut EventEmitter) {
        e.group_start(None, false);
        e.token(TokenKind::SELECT_KW);
        e.space();
        self.target_list
            .iter()
            .for_each(|target| target.to_tokens(e));
        e.space();
        self.from_clause.iter().for_each(|from| from.to_tokens(e));
        e.group_end();
    }
}
