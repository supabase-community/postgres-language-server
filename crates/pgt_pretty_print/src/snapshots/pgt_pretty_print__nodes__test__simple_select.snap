---
source: crates/pgt_pretty_print/src/nodes.rs
expression: emitter.events
snapshot_kind: text
---
[
    GroupStart {
        kind: SelectStmt,
    },
    Token(
        SELECT_KW,
    ),
    IndentStart,
    Line(
        SoftOrSpace,
    ),
    GroupStart {
        kind: ResTarget,
    },
    GroupStart {
        kind: ColumnRef,
    },
    Token(
        IDENT(
            "public",
        ),
    ),
    Token(
        DOT,
    ),
    Token(
        IDENT(
            "t",
        ),
    ),
    Token(
        DOT,
    ),
    Token(
        IDENT(
            "a",
        ),
    ),
    GroupEnd,
    Space,
    Token(
        AS_KW,
    ),
    Space,
    Token(
        IDENT(
            "y",
        ),
    ),
    GroupEnd,
    Token(
        COMMA,
    ),
    Line(
        SoftOrSpace,
    ),
    GroupStart {
        kind: ResTarget,
    },
    GroupStart {
        kind: ColumnRef,
    },
    Token(
        IDENT(
            "b",
        ),
    ),
    GroupEnd,
    Space,
    Token(
        AS_KW,
    ),
    Space,
    Token(
        IDENT(
            "z",
        ),
    ),
    GroupEnd,
    Token(
        COMMA,
    ),
    Line(
        SoftOrSpace,
    ),
    GroupStart {
        kind: ResTarget,
    },
    GroupStart {
        kind: ColumnRef,
    },
    Token(
        IDENT(
            "c",
        ),
    ),
    GroupEnd,
    GroupEnd,
    IndentEnd,
    Line(
        SoftOrSpace,
    ),
    Token(
        FROM_KW,
    ),
    Line(
        SoftOrSpace,
    ),
    IndentStart,
    GroupStart {
        kind: RangeVar,
    },
    Token(
        IDENT(
            "t",
        ),
    ),
    GroupEnd,
    IndentEnd,
    Line(
        SoftOrSpace,
    ),
    Token(
        WHERE_KW,
    ),
    Space,
    GroupStart {
        kind: AExpr,
    },
    GroupStart {
        kind: ColumnRef,
    },
    Token(
        IDENT(
            "id",
        ),
    ),
    GroupEnd,
    Space,
    Token(
        IDENT(
            "=",
        ),
    ),
    Space,
    GroupStart {
        kind: AExpr,
    },
    Space,
    Token(
        IDENT(
            "@",
        ),
    ),
    Space,
    GroupStart {
        kind: ColumnRef,
    },
    Token(
        IDENT(
            "id",
        ),
    ),
    GroupEnd,
    GroupEnd,
    GroupEnd,
    Token(
        SEMICOLON,
    ),
    GroupEnd,
]
